crates/astria-sequencer-relayer/src/relayer/celestia_client/celestia_keys.rs

```diff
@@ -0,0 +1,145 @@
+use std::{
+    fmt::{
+        self,
+        Debug,
+        Formatter,
+    },
+    fs,
+    path::Path,
+};
+
+use k256::ecdsa::{
+    signature::Signer,
+    Signature,
+};
+use tendermint::{
+    account::Id as AccountId,
+    private_key::Secp256k1 as SigningKey,
+    public_key::Secp256k1 as VerificationKey,
+};
+use thiserror::Error;
+use tracing::instrument;
+
+/// Errors which can be returned when parsing a Celestia signing key file.
+///
+/// Note that the path to the file is not included in any of the variants in order to avoid
+/// displaying potentially sensitive path information in logs.
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub(crate) enum Error {
+    /// Failed to read the given file.
+    #[error("failed to read celestia signing key file")]
+    ReadFile(#[from] std::io::Error),
+    /// Failed to decode the file contents from hex.
+    #[error("failed to hex-decode celestia signing key")]
+    DecodeFromHex(#[from] FromHexError),
+    /// The file doesn't contain a valid signing key.
+    #[error("invalid signing key")]
+    InvalidSigningKey,
+}
+
+/// An error while decoding a hex string.
+#[derive(Error, Debug)]
+#[error(transparent)]
+pub(crate) struct FromHexError(#[from] hex::FromHexError);
+
+#[derive(Clone)]
+pub(crate) struct CelestiaKeys {
+    /// The celestia account address; defined as SHA256(public key)[..20].
+    pub(crate) address: AccountId,
+
+    /// The signing (secret) key.
+    pub(crate) signing_key: SigningKey,
+
+    /// The verifying (public) key.
+    pub(crate) verification_key: VerificationKey,
+}
+
+impl CelestiaKeys {
+    /// Constructs `CelestiaKeys` from the given file.
+    ///
+    /// The file should be a hex-encoded secp256k1 secret key, such as could be output via
+    /// `celestia-appd keys export <keyname> --keyring-backend=... --home=... --unsafe
+    /// --unarmored-hex`
+    #[instrument(skip_all, fields(path = % path.as_ref().display(), err))]
+    pub(crate) fn from_path(path: impl AsRef<Path>) -> Result<Self, Error> {
```

Comment:
would be good to note somewhere that the celestia key file is a plain hex-encoded private key, not a keyfile generated by celestia-app

Reply:
Done in [1a0b457](https://github.com/astriaorg/astria/pull/963/commits/1a0b45700ab4da2f7b29a7e1ca37e92173dde0b8).